pkg/build/controller/build/build_controller.go

```diff
@@ -834,4 +837,27 @@ func setBuildCompletionTimestampAndDuration(build *buildapi.Build, podStartTime
 		update.setStartTime(*startTime)
 	}
 	update.setDuration(now.Rfc3339Copy().Time.Sub(startTime.Rfc3339Copy().Time))
+
+	if pod != nil && len(pod.Status.ContainerStatuses) != 0 && pod.Status.ContainerStatuses[0].State.Terminated != nil {
+		msg := pod.Status.ContainerStatuses[0].State.Terminated.Message
+		if len(msg) != 0 {
+			parts := strings.Split(msg, "\n")
```

Comment:
I think this is a little on the baroque side.  In our case Message is already bounded to min(2KB, 80 lines) (see `readLastStringFromContainerLogs`).    Surely doing nothing more, or snipping the last n bytes or lines will do?